go/libkb/globals.go

```diff
@@ -451,3 +459,17 @@ func (g *GlobalContext) GetRuntimeDir() string {
 func (g *GlobalContext) GetRunMode() RunMode {
 	return g.Env.GetRunMode()
 }
+
+func (g *GlobalContext) Now() time.Time {
+	if g.Clock == nil {
+		return time.Now()
+	}
+	return g.Clock.Now()
+}
+
+func (g *GlobalContext) After(d time.Duration) <-chan time.Time {
+	if g.Clock == nil {
+		return time.After(d)
+	}
+	return g.Clock.After(d)
+}
```

Comment:
Not crazy about adding Now() and After() to GlobalContext.

Can't you use the clock interface you added below?  LoginProvision could have a clock field that it would pass along to  newTimedGenericKey...